code/game/jobs/job/outsider/representative.dm

```diff
@@ -76,104 +79,29 @@
-	)
-
-	implants = null
-
-
-/datum/outfit/job/representative/ceti/get_objectives(mob/living/carbon/human/H, mission_level)
-	var/rep_objectives
-
-	switch(mission_level)
-		if("high")
-			rep_objectives = pick("Compile and report and audit [rand(1,3)] suspicious indivduals who might be spies or otherwise act hostile against the Republic",
-							"Have [rand(2,6)] crewmembers sign a pledge of loyalty to the Republic")
-
-		if("medium")
-			rep_objectives = pick("Sell [rand(2,5)] Tau Ceti residence visas to foreign employees, 2000 credits each.",
-							"Convince [rand(3,6)] crewmembers of Tau Ceti superiority over the Sol Alliance")
-		else
-			rep_objectives = pick("Run a questionnaire on Tau Ceti citizens' views on synthetic citizenship",
-							"Run a questionnaire on Tau Ceti citizens' views on vaurca citizenship")
-
-
-	return rep_objectives
-
-/datum/outfit/job/representative/sol
-	name = "Sol Consular Officer"
+/datum/outfit/job/representative/consular
+	name = "Consular Officer"
 
 	uniform = /obj/item/clothing/under/suit_jacket/navy
 	head = null
 	suit = null
 	backpack_contents = list(
-		/obj/item/clothing/accessory/sol_pin = 1,
-		/obj/item/weapon/storage/box/sol_visa = 1,
 		/obj/item/device/camera = 1,
-		/obj/item/weapon/gun/projectile/pistol/sol = 1
+		/obj/item/weapon/gun/energy/pistol = 1
 	)
-
 	implants = null
 
-/datum/outfit/job/representative/sol/get_objectives(mob/living/carbon/human/H, mission_level)
+/datum/outfit/job/representative/consular/get_objectives(var/mob/living/carbon/human/H, var/mission_level)
 	var/rep_objectives
 
-	switch(mission_level)
-		if("high")
-			rep_objectives = pick("Collect evidence of NanoTrasen being unfair or oppressive against Solarian employees, to be used as leverage in future diplomatic talks",
-							"Convince [rand(1,3)] solarian employees to apply for the solarian armed forces")
-
-		if("medium")
-			rep_objectives = pick("Have [rand(2,5)] amount of Sol citizens write down their grievances with the company, and present the report to station command",
-							"Convince [rand(3,6)] human crewmembers to apply for solarian citizenship")
-		else
-			rep_objectives = pick("Collect [rand(3,7)] pictures of secure station areas",
-							"Convince station command to turn a solarian crewmember's sentence into a fine.")
-
-	return rep_objectives
-
-/datum/outfit/job/representative/pra
-	name = "PRA Consular Officer"
-
-	uniform = /obj/item/clothing/under/suit_jacket/checkered
-	head = /obj/item/clothing/head/fedora
-	suit = null
-	backpack_contents = list(
-		/obj/item/clothing/accessory/hadii_pin = 1,
-		/obj/item/weapon/storage/box/hadii_card = 1,
-		/obj/item/weapon/storage/box/hadii_manifesto = 1,
-		/obj/item/weapon/gun/projectile/pistol = 1
-	)
-
-	implants = null
-
-/datum/outfit/job/representative/pra/get_objectives(mob/living/carbon/human/H, mission_level)
-	var/rep_objectives
+	for (var/datum/citizenship/citizenship in SSrecords.citizenships)
```

Comment:
This entire loop is unnecessary. Why not just do:
```dm
	rep_objectives = H?.citizenship.get_objectives(mission_level)
```

Reply:
Not really possible due to citizenship not being a datum reference here.